pkg/util/client/leaderelectionconfig/config.go

```diff
@@ -69,5 +70,6 @@ func BindFlags(l *options.LeaderElectionConfiguration, fs *pflag.FlagSet) {
 		"of a leadership. This is only applicable if leader election is enabled.")
 	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
-		"leader election. Supported options are `endpoints` (default) and `configmap`.")
+		"leader election. Supported options are 'endpoints', 'configmaps', "+
+		"'leases', 'endpointsleases' (default) and 'configmapsleases'.")
```

Comment:
```suggestion
		"'leases', 'endpointsleases' and 'configmapsleases'.")
```


The default value is already added/documented by the underlying library. You can check this out by running:

```
docker run eu.gcr.io/gardener-project/gardener/machine-controller-manager:v0.42.0 --help

      --leader-elect-resource-lock endpoints                    The type of resource object that is used for locking during leader election. Supported options are endpoints (default) and `configmap`. (default "endpoints")
```

There is no need to document it one more time in the doc string of this flag.